### PR TITLE
feat(eslint-config): enable "no-return-await" rule

### PR DESCRIPTION
**Pull request by bajtos**
See https://eslint.org/docs/rules/no-return-await:

> Since the return value of an async function is always wrapped in
> `Promise.resolve, `return await` doesn’t actually do anything except
> add extra time before the overarching `Promise` resolves or rejects.

- The first commit modifies `@loopback/eslint-config`. This is a breaking change.
- The second commits fixes violations in our code base. This is a backwards-compatible chore.
- The third commit adds a new task to VSCode config to configure eslint problem matcher for `npm run lint:fix`.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ describe('HttpCachingProxy', () => {
   let proxy: HttpCachingProxy;
   after(stopProxy);
 
-  beforeEach('clean cache dir', async () => await rimraf(CACHE_DIR));
+  beforeEach('clean cache dir', async () => rimraf(CACHE_DIR));
```
Comment (nabdelgadir) [inline packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts]: Why is this one removed?

Comment (bajtos) [reply packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts]:
```ts
async () => await rimraf(CACHE_DIR)
```

The code above is essentially the same as the following:

```ts
(async function () { return await rimraf(CACHE_DIR); }).bind(this)
```

Comment (nabdelgadir) [reply packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts]: Ah, thank you.

Comment (raymondfeng) [reply packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts]: In this case, `async` is not required either. We can do `beforeEach('clean cache dir', () => rimraf(CACHE_DIR));` as Mocha supports a function that returns a Promise.

File: docs/site/BelongsTo-relation.md [hunk 2]
```diff
@@ -188,7 +188,7 @@ export class OrderController {
   async getCustomer(
     @param.path.number('id') orderId: typeof Order.prototype.id,
   ): Promise<Customer> {
-    return await this.orderRepository.customer(orderId);
+    return this.orderRepository.customer(orderId);
```
Comment (raymondfeng) [inline docs/site/BelongsTo-relation.md]: Now no `await` is used. Do we want to remove `async` keyword for the method signature?

Comment (raymondfeng) [reply docs/site/BelongsTo-relation.md]: Maybe keep `async` so that the code can be modified to use `await` before the `return` statement.

**Pull request merged**